docs/msi_v2/caching_strategy.md

```diff
@@ -0,0 +1,66 @@
+# Managed Identity v2 (Attested TB) — Resilience & Caching Plan
+
+## TL;DR
+We reduce cold-start latency and dependency risk for MSI v2 by caching safe, long-lived artifacts, coordinating renewal across processes, and keeping the hot path in memory. **MAA is used only to (re)issue the binding certificate**; bound AT acquisition relies on that cert. Result: fewer failures, less churn, smoother CX.
+
+---
+
+## Problem
+- Cold starts/reboots trigger extra external calls (MAA → IMDS → eSTS).
+- OS certificate store I/O can contend under load.
+- Multiple processes may race to re-issue binding certificates.
+- We want resilience to **MAA** issues and predictable **cert renewal**.
+
+---
+
+## Solution (What’s Changing)
+1. **Probe once** (link-local) to detect **MSI v2** → cache result **in-proc**.
+2. Treat the **binding certificate** (from IMDS `/issuecredential`) as the **primary anchor** (~7-day validity); use it to get ATs.
+3. **Proactive renewal at half-life (+ small jitter)** to rotate well before expiry.
+4. **Single-writer coordination** so only one process issues/renews; others reuse the same cert.
+5. **MAA token** is used **only** for issuance/renewal; short-lived cache to prevent attestations calls.
+
+---
+
+## Call Sequence (cold start)
+```
+Call 0 (local): Probe IMDS v2 → cache MSI source (V2/V1)
+1       (local): Create KeyGuard key (per reboot)
+2       (external): Get MAA token  // only for (re)issuing cert
+3       (local): IMDS /issuecredential → binding cert + metadata
+4       (external): eSTS-R → bound AT (mtls_pop/bearer) using client mTLS
+5       (external): Call resource with bound AT + client mTLS
+```
+---
+
+## Cache & Renewal Matrix
+
+| Item | Scope | Where | TTL | Notes |
+|---|---|---|---|---|
+| **MSI v2 probe result** | Per process | In-proc static | Process lifetime | NO changes needed here |
+| **MAA token** | Per **keyHandle** | small file cache | ≤ JWT `exp` (~8h) | Only for cert issuance; evict on reboot/policy change/attest fail; refresh half-life + jitter |
```

Comment:
How are policy changes detected? Is it polled, pushed, or inferred from failures?